lib/rules/no-smart-quotes.js

```diff
@@ -41,29 +49,47 @@ module.exports = {
                     return;
                 }
 
-                const match = node.raw.match(/^(.*(“|”|‘|’).*)$/);
-
-                if (match) {
-                    const offendingPunctuation = match[2];
-                    const matchedString = match[1];
-                    const alternate = {
-                        '“': '"',
-                        '”': '"',
-                        '‘': "'",
-                        '’': "'",
-                    } [offendingPunctuation];
-
-                    if (typeof matchedString !== 'undefined' && typeof offendingPunctuation !== 'undefined') {
-                        const message = `Strings must use straight quotes in: ${ matchedString}. Use ${alternate} instead.`;
-                        context.report({
-                            node,
-                            message: message,
-                            data: {
-                                offendingPunctuation
-                            }
-                        });
-                    }
+                const matches = Array.from(matchAll(node.raw, SMART_QUOTES_REGEX));
+
+                if (matches.length === 0) {
+                    return;
                 }
+
+                context.report({
+                    node,
+                    message: "Strings must use straight quotes.",
```

Comment:
I simplified the message since it is fixable now. This seems to be consistent with other fixable rules, and since I'm matching multiple things at once it doesn't make since to suggest a single character replacement.